tagstudio/tests/core/test_tags.py

```diff
@@ -1,18 +1,161 @@
+        assert test_tag1.debug_name() == "tag1 Name (ID: 1)"
+        assert test_tag2.debug_name() == "tag2 Name (ID: 2)"
+
+    def test_display_name_no_shorthand(self, mock_library, test_tag1, test_tag2):
+        test_tag2.shorthand = ""
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == "tag1 Name (tag2 Name)"
+
+    def test_display_name_with_shorthand(self, mock_library, test_tag1, test_tag2):
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == "tag1 Name (tag2)"
+
+    def test_display_name_no_subtags(self, mock_library, test_tag1):
+        test_tag1.subtag_ids = []
+        result = test_tag1.display_name(mock_library)
+        assert result == "tag1 Name"
+'''
+    #This probably isn't how we want display_names to work. But these tests pass if uncommented
+    def test_display_name_no_name(self, mock_library, test_tag1, test_tag2):
+        test_tag2.name = ""
+        test_tag1.name = ""
+        test_tag2.shorthand = ""
+        test_tag1.shorthand = ""
+
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == " ()"
+
+    def test_display_name_no_name_no_subtag_ids(self, mock_library, test_tag1):
+        test_tag1.name = ""
+        test_tag1.shorthand = ""
+        test_tag1.subtag_ids = []
+
+        result = test_tag1.display_name(mock_library)
+        assert result == ""
+
+    def test_display_name_no_name_with_subtag_name(self, mock_library, test_tag1, test_tag2):
+        test_tag1.name = ""
+        test_tag1.shorthand = ""
+        test_tag1.subtag_ids = [2]
+
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == " (tag2)"
+
+    def test_display_name_no_name_with_subtag_name_no_shorthand(self, mock_library, test_tag1, test_tag2):
+        test_tag1.name = ""
+        test_tag1.shorthand = ""
+        test_tag1.subtag_ids = [2]
+        test_tag2.shorthand = ""
+
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == " (tag2 Name)"
+
+    def test_display_name_2_subtags_no_shorthand(self, mock_library, test_tag1, test_tag2, test_tag3):
+        test_tag2.shorthand = ""
+        test_tag3.shorthand = ""
+        test_tag1.subtag_ids = [2,3]
 
-def test_construction():
-    tag = Tag(
-        id=1,
-        name="Tag Name",
-        shorthand="TN",
-        aliases=["First A", "Second A"],
-        subtags_ids=[2, 3, 4],
-        color="",
-    )
-    assert tag
+        mock_library.get_tag.return_value = test_tag2
+        result = test_tag1.display_name(mock_library)
+        assert result == "tag1 Name (tag2 Name)"
+'''
 
+if __name__ == "__main__":
+    pytest.main()
```

Comment:
I dont think this is needed.

Reply:
You're right it isn't needed. I've removed it in the latest commits.